rollup-sequencer/src/index.ts

```diff
@@ -1,151 +1,73 @@
+import { getPublicClient } from "./viem/client.js";
+import { webSocketTransport } from "./viem/transport.js";
 
 async function main() {
 	let lastSubmitted = "";
-	const abi = rolldownAbi.abi;
-	const publicClient = createPublicClient({
-		transport: webSocket(process.env.ETH_CHAIN_URL, {
-			retryCount: 5,
-		}),
-	});
-
-	while (true) {
-		try {
-			const data = (await publicClient.readContract({
-				address: mangataContractAddress,
-				abi: abi,
-				functionName: "getUpdateForL2",
-			})) as any;
-			console.log(data);
-			break;
-		} catch (e) {
-			console.log(`${mangataContractAddress} contract not found`);
-			await sleep_ms(1000);
-		}
-	}
-
-	const api = await Mangata.instance([process.env.MANGATA_NODE_URL!]).api();
-	await api.isReady;
-
-	const keyring = new Keyring({ type: "ethereum" });
-	const collator = keyring.addFromSeed(hexToU8a(process.env.MNEMONIC!));
-
-	await api.derive.chain.subscribeNewHeads(async (header) => {
-		const apiAt = await api.at(header.hash);
-		console.log(`block #${header.number} was authored by ${header.author}`);
-
-		try {
-			const latestBlockNumber = await publicClient.getBlockNumber();
-			const delayedBlockNumber = latestBlockNumber - BigInt(blockNumberDelay);
 
-			console.log(`Latest Block Number: ${latestBlockNumber.toString()}`);
-			console.log(`Delayed Block Number:  ${delayedBlockNumber.toString()}`);
+	const publicClient = getPublicClient({ transport: webSocketTransport });
 
-			const data = (await publicClient.readContract({
-				address: mangataContractAddress,
-				abi: abi,
-				functionName: "getUpdateForL2",
-				blockNumber: delayedBlockNumber,
-			})) as any;
+	const api = await getApi(MANGATA_NODE_URL);
 
-			console.log(util.inspect(data, { depth: null }));
+	await initReadContractWithRetry(publicClient);
 
-			// @ts-ignore
-			const encodedData = encodeAbiParameters(
-				abi.find((e: any) => e!.name === "getUpdateForL2")!.outputs!,
-				[data],
-			);
+	await api.derive.chain.subscribeNewHeads(async (header: HeaderExtended) => {
+		const collator = getCollator("ethereum", MNEMONIC);
 
-			const nativeL1Update = await api.rpc.rolldown.get_native_l1_update(
-				encodedData.substring(2),
-			);
+		print(`block #${header.number} was authored by ${header.author}`);
+		const {isSequencerSelected, hasSequencerRights, selectedSequencer} = await getSelectedSequencerWithRights(api, collator.address, header.hash)
+		if (isSequencerSelected && hasSequencerRights) {
+			print(`Sequencer selected: ${selectedSequencer}`)
+			try {
 
-			if (lastSubmitted !== keccak256(encodedData)) {
-				await signTx(
+				const { encodedData, nativeL1Update } = await processDataForL2Update(
 					api,
-					api.tx.rolldown.updateL2FromL1(nativeL1Update.unwrap()),
-					collator,
+					publicClient,
 				);
-				lastSubmitted = keccak256(encodedData);
-			} else {
-				console.log(`L1Update was already submitted ${encodedData}`);
+
+				if (lastSubmitted !== keccak256(encodedData)) {
+					await signTx(
```

Comment:
so it turns out we need request splitting and filtering

https://github.com/mangata-finance/eigen-layer-monorepo/blob/rollup-testnet/rollup-sequencer/src/index.ts#L212-L230

esseitnally there should be new env Variable LIMIT.

once sequencer gets the update from l1, it should filter new requests(quering l2 storage as in link below) and only provide up to N following updates by removing the rest 

https://github.com/mangata-finance/eigen-layer-monorepo/blob/rollup-testnet/rollup-sequencer/src/index.ts#L212-L230